lib/SIL/SILVerifier.cpp

```diff
@@ -2675,14 +2675,26 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
               "no arguments");
   }
 
+  bool verifyBranchArgs(SILValue branchArg, SILArgument *bbArg) {
+    // NOTE: IRGen currently does not support the following method_inst
+    // variants as branch arguments.
+    // Once this is supported, the check can be removed.
+    require(
+        !isa<WitnessMethodInst>(branchArg) &&
+            !(isa<MethodInst>(branchArg) &&
+              cast<MethodInst>(branchArg)->getMember().getDecl()->isObjC()),
```

Comment:
Instead of checking whether the decl is `@objc`, you should check whether the member `isForeign`. Referring to the Swift entry point of an `@objc` method should be OK.